### PR TITLE
[MRG+1] Duplicate channel names appended with a running number.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/4305

### CODE REVIEW

File: mne/io/tests/test_meas_info.py [hunk 1]
```diff
@@ -339,6 +339,15 @@ def test_check_consistency():
...
     assert_raises(RuntimeError, info2._check_consistency)
 
+    # Duplicates appended with running numbers
+    info2['ch_names'][2] = 'b'
+    with warnings.catch_warnings(record=True) as w:
+        info2._check_consistency()
+    assert_equal(len(w), 1)
+    assert_true(all(ww.message.message.startswith('Channel names are not '
+                                                  'unique') for ww in w))
+    assert_true(info2['ch_names'][2] == 'b-1')
```
Comment (agramfort) [inline mne/io/tests/test_meas_info.py]: can you test that you have now b-0 and b-1?

can you check also that it works with 2 duplicates channels?

**Comment by agramfort** [comment]
LGTM
+1 for merge when CIs are happy

File: mne/io/tests/test_meas_info.py [hunk 2]
```diff
@@ -339,6 +339,14 @@ def test_check_consistency():
     info2['chs'][2]['ch_name'] = 'b'
     assert_raises(RuntimeError, info2._check_consistency)
 
+    # Duplicates appended with running numbers
+    with warnings.catch_warnings(record=True) as w:
+        info3 = create_info(ch_names=['a', 'b', 'b', 'c', 'b'], sfreq=1000.)
+    assert_equal(len(w), 1)
+    assert_true(all(ww.message.message.startswith('Channel names are not '
```
Comment (Eric89GXL) [inline mne/io/tests/test_meas_info.py]: `.message.message`?

https://travis-ci.org/mne-tools/mne-python/jobs/244504640

**Comment by agramfort** [comment]
travis is not happy but I did not look why

besides LGTM

**Comment by jaeilepp** [comment]
I rebased. Should be good shortly.

**Comment by agramfort** [comment]
time exceeded on two builds. Failure is unrelated. Merging.

thx @jaeilepp

**Pull request merged**